web-bundle/src/main/java/com/graphhopper/resources/MapboxResponseConverter.java

```diff
@@ -0,0 +1,324 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.resources;
+
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.graphhopper.GHResponse;
+import com.graphhopper.PathWrapper;
+import com.graphhopper.http.WebHelper;
+import com.graphhopper.util.*;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.UUID;
+
+public class MapboxResponseConverter {
+
+    // TODO: this is already in GraphHopper, but getting it from there is not possible easily?
+    private static final TranslationMap trMap = new TranslationMap().doImport();
+
+
+    /**
+     * Converts a GHResponse into Mapbox compatible json
+     */
+    public static ObjectNode convertFromGHResponse(GHResponse ghResponse, Locale locale) {
+        ObjectNode json = JsonNodeFactory.instance.objectNode();
+
+        if (ghResponse.hasErrors())
+            throw new IllegalStateException("Cannot convert an errornous ghResponse");
```

Comment:
This should be possible. I mean, what if the route is not found or you want to go to the US from Europe. We have to support error JSON too.

Reply:
Yes, it is :). It's just a different method, see `MapboxResponseConverter#convertFromGHResponseError`. Since we have to change the http response status to 422 in `MapboxResource` (at line ~137), I thought it would be a bit cleaner to have the two methods separated. It would be easily mergeable though. WDYT?

Reply:
I see. Then the error message needs to be changed to "should be handled before in MapboxResource"?

Reply:
Ok, sure :)